frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -104,10 +104,12 @@ const getMetadata = (formData: DeployImageFormData) => {
     name,
     isi: { image },
     labels: userLabels,
+    registry,
     imageStream: { image: imgName, tag: imgTag, namespace: imgNamespace },
   } = formData;
-
-  const defaultLabels = getAppLabels(name, application, imgName || undefined, imgTag, imgNamespace);
+  const imgStreamName =
+    registry === RegistryType.Internal ? getRuntime(image.metadata.labels) : imgName;
```

Comment:
@karthikjeeyar `imgName` is set only in case of internal registry so anyways if external registry is selected then imgName is going to be empty. So you can simply do this - 
```suggestion
  const imgStreamName =
    registry === RegistryType.Internal && getRuntime(image.metadata.labels)
```
Also this should be true for external registry since currently we are not adding runtime/name labels to the resources created using it as `imgName` is empty in its case. But we can have a separate bug for that.

Reply:
updated it.